src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -66,6 +71,20 @@ func (gc *GarbageCollector) Validate(params job.Parameters) error {
 }
 
 // Run implements the interface in job/Interface
+// The workflow of GC is:
+// 1, set harbor to readonly
+// 2, select the candidate artifacts from Harbor DB.
+// 3, call registry API(--delete-untagged=false) to delete manifest bases on the results of #2
+// 4, clean keys of redis DB of registry, clean artifact trash and untagged from DB.
+// 5, roll back readonly.
+// More details:
+// 1, why disable delete untagged when to call registry API
```

Comment:
A reason that isn't so corner case is that we introduce `Harbor tag` in v2.0, Harbor tag is stored in database and there is no corresponding data in registry.
Please add this as well in case we forget in future

Reply:
will do